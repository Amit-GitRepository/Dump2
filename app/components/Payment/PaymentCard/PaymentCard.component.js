/* PaymentCard: includes card form and transaction amount*/

import CardSelection from '../CardSelection/CardSelection.component';
import Collapsible from 'react-native-collapsible';
import PaymentCardForm from '../PaymentCardForm/PaymentCardForm.component';
import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './PaymentCard.style';
import {AmountText, Button, Icon, ISText, KeyboardScrollView, Modal, WrapperCard} from '../../Shared';
import {CARD_TYPE, MAXIMUM_SAVE_CARD_COUNT} from '../../../config/payment.config';
import {findIndex, isEmpty, noop, sumBy} from 'lodash';
import {REMOVE_EXISTING_CARD} from '../../../redux/actions/index.actions';
import {translate} from '../../../language/i18n/helper';
import {View} from 'react-native';

const generateCardDeleteMessage = (payload) => ({
  title: translate('PAYMENT__CARD_DELETE_MODAL_HEADER'),
  body: translate('PAYMENT__CARD_DELETE_MODAL_DESC'),
  buttons: [
    {
      title: translate('BUTTON__NO')
    },
    {
      title: translate('BUTTON__YES'),
      actionType: REMOVE_EXISTING_CARD,
      nextAction: payload
    }
  ]
});

class PaymentCard extends Component {

  state = {
    selectedCard: {
      index: null,
      type: null,
      id: null
    },
    formData: {},
    isOtherCollapsed: true,
    replaceCardModalVisiblity: false,
    replaceCardId: ''
  }

  componentWillReceiveProps = (nextProps) => {
    const {cards} = this.props.savedCards;
    const {cards: nextCards} = nextProps.savedCards;
    const {selectedCard} = this.state;
    const {index, type, id} = selectedCard;
    // Check if the cards got deleted
    if (cards.length !== nextCards.length && index) {
      // check if the existing selected card got deleted
      if (type === CARD_TYPE.SAVED && findIndex(cards, {id}) > -1) {
        this.setState({selectedCard: {index: null, type: null, id: null}});
      } else {
        this.setState({selectedCard: {...selectedCard, index: index - 1}});
      }
    }
  }

  deleteExistingCard = (id) => {
    const {showPopup} = this.props;
    const message = generateCardDeleteMessage(id);
    showPopup(message);
  }

  onCardSelect = (index, type, id) => (radioVal, radioButtonOldStatus) => {
    if (radioButtonOldStatus) { // if radioButtonOldStatus was already true, don't do anything
      return;
    }
    const isOtherCollapsed = (type === CARD_TYPE.OTHER) ? !this.state.isOtherCollapsed : true;
    this.setState({selectedCard: {index, type, id}, isOtherCollapsed, formData: {}});
  }

  onFormDateChanged = (type) => (value) => {
    const newFormData = {...this.state.formData, [type]: value};
    this.setState({formData: newFormData});
  }

  onSubmit = () => {
    const {selectedCard: {type, index}, formData} = this.state;
    const {userProfile, payForNewCustomer, payForExistingCustomer, preLoginPayment,
      paymentItems, savedCards: {customerId, cards = []}} = this.props;
    if (!isEmpty(userProfile)) {
      // check for existing customer
      if (customerId) {
        if (formData.saveCard && cards.length >= MAXIMUM_SAVE_CARD_COUNT) { // if saved card limit is exceeded, show modal
          this.setState({replaceCardModalVisiblity: true});
        } else {
          payForExistingCustomer({type, customerId, paymentItems, formData, selectedCard: cards[index]});
        }
      } else {
        payForNewCustomer({paymentItems, formData});
      }
    } else {
      preLoginPayment({paymentItems, formData});
    }
  }

  setReplaceCardId = (replaceCardId) => () => this.setState({replaceCardId})
  closeModal = () => this.setState({replaceCardModalVisiblity: false})
  replaceCardAndPay = () => {
    const {formData, replaceCardId} = this.state;
    const {paymentItems, savedCards: {customerId}, replaceCard} = this.props;
    const cardData = {
      id: replaceCardId,
      customerId,
      paymentItems,
      formData
    };
    replaceCard(cardData);
    this.setState({replaceCardModalVisiblity: false});
  }

  render () {
    const {userProfile, onFormSubmit, cardLimit, savedCards, paymentItems} = this.props;
    const {selectedCard, isOtherCollapsed, formData, replaceCardModalVisiblity, replaceCardId} = this.state;
    const noOfSavedCards = savedCards.cards.length;
    let amount = sumBy(paymentItems, (item) => item.amount);
    amount = amount.toFixed(2);
    return (
      <KeyboardScrollView contentContainerStyle={styles.contentContainer}>
        <WrapperCard contentContainerStyles={styles.container}>
          <View style={styles.topSection}>
            <ISText type='BOLD' style={styles.amountText}>{translate('PAYMENT__AMOUNT')}</ISText>
            <AmountText value={Number(amount)} isTotalText={true}/>
          </View>
          <View style={styles.textBar}>
            <ISText type='SEMIBOLD' style={styles.textBarTitle}>{translate('PAYMENT__CARD_DETAILS')}</ISText>
          </View>
          {
            noOfSavedCards > 0 ? savedCards.cards.map((card, index) =>
              [
                <CardSelection key={`SaveCards_${index}`} isSelected={selectedCard.index === index} deleteCard={this.deleteExistingCard} cardDetails={card} onSelect={this.onCardSelect(index, CARD_TYPE.SAVED, card.id)}/>,
                (noOfSavedCards - 1 === index) && <View key={`SaveCards_${index + 1}`}>
                  <CardSelection isSelected={selectedCard.index === index + 1} cardDetails={{name: 'Other'}} onSelect={this.onCardSelect(index + 1, CARD_TYPE.OTHER)} isLast={true}/>
                  <Collapsible collapsed={isOtherCollapsed}>
                    <PaymentCardForm onFormSubmit={onFormSubmit} amount={amount} cardLimit={cardLimit} formData={formData} onDataChanged={this.onFormDateChanged}/>
                  </Collapsible>
                </View>
              ])
              : <PaymentCardForm canSaveCards={!isEmpty(userProfile)} onFormSubmit={onFormSubmit} amount={amount} noOfSavedCards={noOfSavedCards} cardLimit={cardLimit} formData={formData} onDataChanged={this.onFormDateChanged}/>
          }
        </WrapperCard>
        <Button text={translate('PAYMENT__PAY_BUTTON', {amount})} disabled={selectedCard.index === null && isEmpty(formData)} style={styles.buttonStyle} onPress={this.onSubmit}/>
        <Modal animationIn={'zoomIn'} animationOut={'zoomOut'} style={styles.modal}
          animationInTiming={400} backdropTransitionInTiming={1000} backdropTransitionOutTiming={1000}
          animationOutTiming={400} isVisible={replaceCardModalVisiblity}>
          <View style={styles.modalContainer}>
            <Icon name='cross' style={styles.cross} onPress={this.closeModal} />
            <ISText type='SEMIBOLD' style={styles.headerText}>{translate('PAYMENT__SAVE_CARD_MODAL_HEADER', {limit: MAXIMUM_SAVE_CARD_COUNT})}</ISText>
            <View style={styles.textBar}>
              <ISText type='SEMIBOLD' style={styles.textBarTitle}>{translate('PAYMENT__SAVE_CARD_MODAL_DESC')}</ISText>
            </View>
            {savedCards.cards.map((card, index) => (
              <CardSelection cardDetails={card} key={index} isLast={index === noOfSavedCards - 1} isSelected={card.id === replaceCardId} onSelect={this.setReplaceCardId(card.id)}/>
            ))}
            <View style={styles.buttonContainer}>
              <Button textType='BOLD' textStyle={{...styles.btnText,  ...styles.secondarybtnText}} type='secondary' touchableStyle={styles.button} text={translate('BUTTON__CANCEL')} onPress={this.closeModal} />
              <View style={styles.btnSeparator} />
              <Button textType='BOLD' textStyle={styles.btnText} type='primary' touchableStyle={styles.button} text={translate('PAYMENT__REPLACE_CARD')} onPress={this.replaceCardAndPay} />
            </View>
          </View>
        </Modal>
      </KeyboardScrollView>);
  }
}
PaymentCard.defaultProps = {
  paymentItems: [],
  onFormSubmit: noop,
  cardLimit: 0,
  payForNewCustomer: noop,
  payForExistingCustomer: noop,
  replaceCard: noop,
  preLoginPayment: noop,
  showPopup: noop,
  savedCards: {},
  userProfile: {}
};
PaymentCard.propTypes = {
  onFormSubmit: PropTypes.func,
  paymentItems: PropTypes.array,
  cardLimit: PropTypes.number,
  payForNewCustomer: PropTypes.func,
  replaceCard: PropTypes.func,
  payForExistingCustomer: PropTypes.func,
  preLoginPayment: PropTypes.func,
  userProfile: PropTypes.object,
  showPopup: PropTypes.func,
  savedCards: PropTypes.object
};
export default PaymentCard;
