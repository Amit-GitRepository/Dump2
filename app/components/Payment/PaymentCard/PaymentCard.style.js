import {colors, fonts} from '../../../themes/constants.styles';
import {contentContainerStyle} from '../../../themes/application.styles';

export default {
  contentContainer: contentContainerStyle,
  container: {
    padding: 10
  },
  modalContainer: {
    backgroundColor: colors.PRIMARY_BG_TEXT_CONTRAST,
    padding: 10,
    borderRadius: 10,
    borderColor: colors.PRIMARY_BG_SEPARATOR,
    borderWidth: 1
  },
  modal: {
    margin: 15
  },
  btnText: {
    fontSize: fonts.FONT_SIZE_MEDIUM
  },
  secondarybtnText: {
    color: colors.PRIMARY_ACTIONABLE
  },
  btnSeparator: {
    width: 10
  },
  headerText: {
    padding: 5,
    textAlign: 'center',
    fontSize: fonts.FONT_SIZE_MEDIUM,
    paddingBottom: 5
  },
  cross: {
    textAlign: 'right',
    fontSize: fonts.FONT_SIZE_MEDIUM,
    color: colors.PRIMARY_DISABLED_BG_TEXT,
    backgroundColor: colors.TRANSPARENT,
    padding: 5
  },
  textBar: {
    backgroundColor: colors.PRIMARY_TEXT_TAB_LABEL,
    padding: 10,
    marginHorizontal: -10
  },
  textBarTitle: {
    textAlign: 'center',
    color: colors.PRIMARY_BG_TEXT_CONTRAST,
    fontSize: fonts.FONT_SIZE_MEDIUM
  },
  topSection: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    paddingBottom: 15,
    alignItems: 'center'
  },
  buttonContainer: {
    flexDirection: 'row',
    paddingVertical: 5
  },
  button: {
    flex: 1,
    borderColor: colors.PRIMARY_ACTIONABLE
  },
  amountText: {
    fontSize: fonts.FONT_SIZE_MEDIUM
  },
  buttonStyle: {
    margin: 10
  }
};
