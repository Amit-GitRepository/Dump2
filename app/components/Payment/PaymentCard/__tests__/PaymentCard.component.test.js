import PaymentCard from '../PaymentCard.component';
import React from 'react';
import renderer from 'react-test-renderer';

describe('PaymentCard component', () => {
  it('renders correctly', () => {
    const savedCards = {cards: []};
    const component = renderer.create(<PaymentCard onFormSubmit={jest.fn} savedCards={savedCards}/>).toJSON();
    expect(component).toMatchSnapshot();
  });
});
