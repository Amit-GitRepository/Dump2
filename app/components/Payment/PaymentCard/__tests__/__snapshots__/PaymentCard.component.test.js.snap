// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentCard component renders correctly 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  contentContainerStyle={
    Object {
      "flexDirection": "column",
      "flexGrow": 1,
      "justifyContent": "space-between",
    }
  }
  contentInset={
    Object {
      "bottom": 0,
    }
  }
  enableAutoAutomaticScroll={true}
  enableResetScrollToCoords={true}
  extraHeight={75}
  extraScrollHeight={0}
  keyboardDismissMode="interactive"
  keyboardOpeningTime={250}
  onScroll={[Function]}
  overScrollMode="auto"
  scrollEventThrottle={0}
  showsVerticalScrollIndicator={true}
>
  <View>
    <View
      style={
        Array [
          Object {
            "marginHorizontal": 8,
          },
          Object {},
        ]
      }
    >
      <View
        elevation={2}
        style={
          Array [
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#B9C1C9",
              "borderRadius": 10,
              "borderWidth": 1,
              "shadowColor": "#303C46",
              "shadowOffset": Object {
                "height": 2,
                "width": 1,
              },
              "shadowOpacity": 0.3,
              "shadowRadius": 1,
            },
            Object {
              "padding": 10,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "justifyContent": "space-between",
              "paddingBottom": 15,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#303C46",
                "fontFamily": "SukhumvitSet-Bold",
                "fontSize": 18,
              }
            }
          >
            Balance required
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "flex-end",
                  "flexDirection": "row",
                },
                Object {},
              ]
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#303C46",
                  },
                  Array [
                    Object {
                      "fontSize": 18,
                      "paddingBottom": 6,
                      "paddingRight": 3,
                    },
                    Object {},
                  ],
                  Object {
                    "fontFamily": "SukhumvitSet-Bold",
                  },
                ]
              }
            >
              à¸¿
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#303C46",
                  },
                  Array [
                    Object {
                      "fontSize": 30,
                    },
                    Object {},
                  ],
                  Object {
                    "fontFamily": "SukhumvitSet-Bold",
                  },
                ]
              }
            >
              0
            </Text>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#303C46",
                  },
                  Array [
                    Object {
                      "color": "#68747E",
                      "fontSize": 18,
                      "paddingBottom": 5,
                    },
                    Object {},
                  ],
                  Object {
                    "fontFamily": "SukhumvitSet-Bold",
                  },
                ]
              }
            >
              .
              00
            </Text>
          </View>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "#303C46",
              "marginHorizontal": -10,
              "padding": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#FFFFFF",
                "fontFamily": "SukhumvitSet-SemiBold",
                "fontSize": 18,
                "textAlign": "center",
              }
            }
          >
            Card details
          </Text>
        </View>
        <View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#303C46",
                  "fontFamily": "SukhumvitSet-SemiBold",
                  "fontSize": 16,
                  "paddingVertical": 10,
                }
              }
            >
              Credit card number
            </Text>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                }
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../app/assets/visa.png",
                  }
                }
                style={
                  Object {
                    "marginHorizontal": 5,
                  }
                }
              />
              <Image
                source={
                  Object {
                    "testUri": "../../../app/assets/mastercard.png",
                  }
                }
                style={
                  Object {
                    "marginHorizontal": 5,
                  }
                }
              />
              <Image
                source={
                  Object {
                    "testUri": "../../../app/assets/jcb.png",
                  }
                }
                style={
                  Array [
                    Object {
                      "marginHorizontal": 5,
                    },
                    Object {
                      "bottom": 1,
                      "marginLeft": -5,
                    },
                  ]
                }
              />
            </View>
          </View>
          <View
            accessibilityLabel={undefined}
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#605E71",
                  "borderRadius": 4,
                  "borderWidth": 2,
                  "flexDirection": "row",
                },
                Object {
                  "borderColor": "#B9C1C9",
                  "borderWidth": 1,
                },
                Object {},
              ]
            }
            testID={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 35,
                  },
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "transparent",
                    "color": "#B9C1C9",
                    "padding": 5,
                  },
                  Object {
                    "fontFamily": "iservice",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
            <TextInput
              allowFontScaling={true}
              defaultValue={undefined}
              keyboardType="numeric"
              maxLength={19}
              onChangeText={[Function]}
              placeholder="Card Number"
              placeholderTextColor="#B9C1C9"
              style={
                Array [
                  Object {
                    "alignSelf": "stretch",
                    "borderRadius": 4,
                    "flex": 1,
                    "fontSize": 16,
                    "paddingHorizontal": 0,
                    "paddingVertical": 10,
                  },
                  Object {},
                ]
              }
              underlineColorAndroid="transparent"
            />
          </View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#303C46",
                "fontFamily": "SukhumvitSet-SemiBold",
                "fontSize": 16,
                "paddingVertical": 10,
              }
            }
          >
            Name on the card
          </Text>
          <View
            accessibilityLabel={undefined}
            style={
              Array [
                Object {
                  "alignSelf": "stretch",
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#605E71",
                  "borderRadius": 4,
                  "borderWidth": 2,
                  "flexDirection": "row",
                },
                Object {
                  "borderColor": "#B9C1C9",
                  "borderWidth": 1,
                },
                Object {},
              ]
            }
            testID={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 35,
                  },
                  Object {
                    "alignSelf": "center",
                    "backgroundColor": "transparent",
                    "color": "#B9C1C9",
                    "padding": 5,
                  },
                  Object {
                    "fontFamily": "iservice",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
            <TextInput
              allowFontScaling={true}
              defaultValue={undefined}
              maxLength={26}
              onChangeText={[Function]}
              placeholder="Cardholder Name"
              placeholderTextColor="#B9C1C9"
              style={
                Array [
                  Object {
                    "alignSelf": "stretch",
                    "borderRadius": 4,
                    "flex": 1,
                    "fontSize": 16,
                    "paddingHorizontal": 0,
                    "paddingVertical": 10,
                  },
                  Object {},
                ]
              }
              underlineColorAndroid="transparent"
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "space-between",
              }
            }
          >
            <View
              style={
                Object {
                  "flex": 1,
                  "paddingRight": 10,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#303C46",
                    "fontFamily": "SukhumvitSet-SemiBold",
                    "fontSize": 16,
                    "paddingVertical": 10,
                  }
                }
              >
                Expiry date
              </Text>
              <View
                accessibilityLabel={undefined}
                style={
                  Array [
                    Object {
                      "alignSelf": "stretch",
                      "backgroundColor": "#FFFFFF",
                      "borderColor": "#605E71",
                      "borderRadius": 4,
                      "borderWidth": 2,
                      "flexDirection": "row",
                    },
                    Object {
                      "borderColor": "#B9C1C9",
                      "borderWidth": 1,
                    },
                    Object {},
                  ]
                }
                testID={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 35,
                      },
                      Object {
                        "alignSelf": "center",
                        "backgroundColor": "transparent",
                        "color": "#B9C1C9",
                        "padding": 5,
                      },
                      Object {
                        "fontFamily": "iservice",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                >
                  
                </Text>
                <TextInput
                  allowFontScaling={true}
                  defaultValue={undefined}
                  maxLength={7}
                  onChangeText={[Function]}
                  placeholder="Date (MM/YYYY)"
                  placeholderTextColor="#B9C1C9"
                  style={
                    Array [
                      Object {
                        "alignSelf": "stretch",
                        "borderRadius": 4,
                        "flex": 1,
                        "fontSize": 16,
                        "paddingHorizontal": 0,
                        "paddingVertical": 10,
                      },
                      Object {},
                    ]
                  }
                  underlineColorAndroid="transparent"
                />
              </View>
            </View>
            <View
              style={
                Object {
                  "flex": 1,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "backgroundColor": "transparent",
                    "color": "#303C46",
                    "fontFamily": "SukhumvitSet-SemiBold",
                    "fontSize": 16,
                    "paddingVertical": 10,
                  }
                }
              >
                Security code
              </Text>
              <View
                accessibilityLabel={undefined}
                style={
                  Array [
                    Object {
                      "alignSelf": "stretch",
                      "backgroundColor": "#FFFFFF",
                      "borderColor": "#605E71",
                      "borderRadius": 4,
                      "borderWidth": 2,
                      "flexDirection": "row",
                    },
                    Object {
                      "borderColor": "#B9C1C9",
                      "borderWidth": 1,
                    },
                    Object {
                      "flex": 1,
                    },
                  ]
                }
                testID={undefined}
              >
                <Text
                  accessible={true}
                  allowFontScaling={false}
                  ellipsizeMode="tail"
                  style={
                    Array [
                      Object {
                        "color": undefined,
                        "fontSize": 35,
                      },
                      Object {
                        "alignSelf": "center",
                        "backgroundColor": "transparent",
                        "color": "#B9C1C9",
                        "padding": 5,
                      },
                      Object {
                        "fontFamily": "iservice",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                    ]
                  }
                >
                  
                </Text>
                <TextInput
                  allowFontScaling={true}
                  defaultValue={undefined}
                  keyboardType="numeric"
                  maxLength={4}
                  onChangeText={[Function]}
                  placeholder="Security Code"
                  placeholderTextColor="#B9C1C9"
                  secureTextEntry={true}
                  style={
                    Array [
                      Object {
                        "alignSelf": "stretch",
                        "borderRadius": 4,
                        "flex": 1,
                        "fontSize": 16,
                        "paddingHorizontal": 0,
                        "paddingVertical": 10,
                      },
                      Object {},
                    ]
                  }
                  underlineColorAndroid="transparent"
                />
              </View>
            </View>
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "justifyContent": "center",
                "marginHorizontal": 30,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#B9C1C9",
                  "fontFamily": "SukhumvitSet-Text",
                  "fontSize": 14,
                  "fontWeight": "600",
                  "marginRight": 10,
                }
              }
            >
              Secured By
            </Text>
            <Image
              resizeMode="contain"
              source={
                Object {
                  "testUri": "../../../app/assets/omise-logo.png",
                }
              }
              style={
                Object {
                  "flex": 0,
                  "width": 60,
                }
              }
            />
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignSelf": "stretch",
          "backgroundColor": "#F93E60",
          "borderRadius": 50,
          "margin": 10,
          "opacity": 1,
          "padding": 10,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <View
        style={
          Object {
            "opacity": 0.7,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "alignSelf": "stretch",
              "flexDirection": "row",
              "justifyContent": "center",
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "backgroundColor": "transparent",
                  "color": "#303C46",
                },
                Array [
                  Object {
                    "color": "#FFFFFF",
                    "fontSize": 16,
                  },
                  Object {},
                ],
                Object {
                  "fontFamily": "SukhumvitSet-Bold",
                },
              ]
            }
          >
            Pay 0.00 Baht
          </Text>
        </View>
      </View>
    </View>
    <Modal
      animationType="none"
      hardwareAccelerated={false}
      onModalHide={[Function]}
      onRequestClose={[Function]}
      transparent={true}
      visible={false}
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        hitSlop={undefined}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Array [
            Object {
              "backgroundColor": "black",
              "bottom": 0,
              "left": 0,
              "opacity": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "backgroundColor": "rgba(48,60,70,0.9)",
              "height": 1334,
              "width": 750,
            },
          ]
        }
        testID={undefined}
      />
      <View
        onModalHide={[Function]}
        pointerEvents="box-none"
        style={
          Array [
            Object {
              "margin": 37.5,
              "transform": Array [
                Object {
                  "translateY": 0,
                },
              ],
            },
            Object {
              "flex": 1,
              "justifyContent": "center",
            },
            Object {
              "margin": 15,
            },
          ]
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "#FFFFFF",
              "borderColor": "#E9E9EA",
              "borderRadius": 10,
              "borderWidth": 1,
              "padding": 10,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={false}
            ellipsizeMode="tail"
            onPress={[Function]}
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontSize": 12,
                },
                Object {
                  "backgroundColor": "transparent",
                  "color": "#B9C1C9",
                  "fontSize": 18,
                  "padding": 5,
                  "textAlign": "right",
                },
                Object {
                  "fontFamily": "iservice",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
              ]
            }
          >
            î¤®
          </Text>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "backgroundColor": "transparent",
                "color": "#303C46",
                "fontFamily": "SukhumvitSet-SemiBold",
                "fontSize": 18,
                "padding": 5,
                "paddingBottom": 5,
                "textAlign": "center",
              }
            }
          >
            Up to 3 cards can be remembered
          </Text>
          <View
            style={
              Object {
                "backgroundColor": "#303C46",
                "marginHorizontal": -10,
                "padding": 10,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "backgroundColor": "transparent",
                  "color": "#FFFFFF",
                  "fontFamily": "SukhumvitSet-SemiBold",
                  "fontSize": 18,
                  "textAlign": "center",
                }
              }
            >
              Select one card below to replace :
            </Text>
          </View>
          <View
            style={
              Object {
                "flexDirection": "row",
                "paddingVertical": 5,
              }
            }
          >
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "stretch",
                  "backgroundColor": "#FFFFFF",
                  "borderColor": "#F93E60",
                  "borderRadius": 50,
                  "borderWidth": 1.5,
                  "flex": 1,
                  "opacity": 1,
                  "padding": 10,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": "stretch",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "color": "#303C46",
                        },
                        Array [
                          Object {
                            "color": "#303C46",
                            "fontSize": 16,
                          },
                          Object {
                            "color": "#F93E60",
                            "fontSize": 18,
                          },
                        ],
                        Object {
                          "fontFamily": "SukhumvitSet-Bold",
                        },
                      ]
                    }
                  >
                    Cancel
                  </Text>
                </View>
              </View>
            </View>
            <View
              style={
                Object {
                  "width": 10,
                }
              }
            />
            <View
              accessibilityComponentType={undefined}
              accessibilityLabel={undefined}
              accessibilityTraits={undefined}
              accessible={true}
              collapsable={undefined}
              hitSlop={undefined}
              isTVSelectable={true}
              nativeID={undefined}
              onLayout={undefined}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Object {
                  "alignSelf": "stretch",
                  "backgroundColor": "#F93E60",
                  "borderColor": "#F93E60",
                  "borderRadius": 50,
                  "flex": 1,
                  "opacity": 1,
                  "padding": 10,
                }
              }
              testID={undefined}
              tvParallaxProperties={undefined}
            >
              <View
                style={
                  Object {
                    "opacity": 1,
                  }
                }
              >
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": "stretch",
                      "flexDirection": "row",
                      "justifyContent": "center",
                    }
                  }
                >
                  <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail"
                    style={
                      Array [
                        Object {
                          "backgroundColor": "transparent",
                          "color": "#303C46",
                        },
                        Array [
                          Object {
                            "color": "#FFFFFF",
                            "fontSize": 16,
                          },
                          Object {
                            "fontSize": 18,
                          },
                        ],
                        Object {
                          "fontFamily": "SukhumvitSet-Bold",
                        },
                      ]
                    }
                  >
                    Replace
                  </Text>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </Modal>
  </View>
</RCTScrollView>
`;
